### ISSUE TITLE
Is there a need to standardize `geom` and `geometry` across ecosystem

### ISSUE THREAD

**Issue by alex-s-gardner** [JuliaGeo/GeoInterface.jl#164]
I'm finding `geom` and `geometry` used interchangeably but this creates some conflicts across package. It seems like some guidance might be needed: 

Documenting some of what I've noticed:

@evetion uses `geom` in GeoDataFrames which then can not be passed directly to @rafaqz Raster.rasterize

@asinghvi17 also uses `geom` in GeometryOps

GeoInterface has a `GeometryCollection` but a function call that is `getgeom` 

Shapefile creates a `geometry` column

**Comment by rafaqz** [comment]
We have `GI.geometrycolumns` to clarify these in many cases. GeometryOps should use it everywhere, we don't use `:geom` ? except in one tutorial because GeoDataFrames.jl does.

`:geometry` is the standard fallback for `GI.geometrycolumns`

 @evetion I noticed this the other day too, is there a reason to still use `:geom` in GeoDataFrames?

**Comment by asinghvi17** [comment]
Using `:geom` for geometry columns started because that was how ArchGDAL returned it, I think.  Users might expect it now though?  I'm not sure how that should be handled.

**Comment by rafaqz** [comment]
We should as much as possible standardise table columns, it's worth the long term simplicity

**Comment by alex-s-gardner** [comment]
e.g.
which one if I'm package blind?
```
GeometryOps.intersects.(geomA[:, :geometry], geomB[:, :geom])
GeometryOps.intersects.(geomA[:, :geo], geomB[:, :geometry])
GeometryOps.intersects.(geomA[:, :geometry], geomB[:, :geometry])
GeometryOps.intersects.(geomA[:, :geom], geomB[:, :geom])
```

or just
```
GeometryOps.intersects.(geomA[:, :geometry], geomB[:, :geometry])
```

**Comment by asinghvi17** [comment]
GeometryOps.intersects.(geomA[:, first(GI.geometrycolumns(geomA))], geomB[:, first(GI.geometrycolumns(geomB))])` is the "correct" solution at this point but it's a mouthful.  If you aren't using GeoDataFrames/ArchGDAL at all for vector features, then `:geometry` is your best bet.

**Comment by rafaqz** [comment]
Yeah... good demonstratiion that `first(GI.geometrycolumns(geomA)` is waay too long to actually use

**Comment by alex-s-gardner** [comment]
Reiterating that standardizing on the geometry column name (either "geometry" or "geom") early could provide long-term benefits. I find myself continually having to check  if "geom" or "geometry"

**Comment by rafaqz** [comment]
Where is it still :geom? This is just GDAL?

There is not much more we can do here, GeoInterface has both a standard name and a function to get non standard names.

Probably an issue at the offending source is more actionable

**Comment by asinghvi17** [comment]
Any non-GeoJSON thing loaded by GDAL has 	`:geom` as a geometry column, this includes everything returned by GeoDataFrames.  That's the primary problem here.

I had two proposals on GeoInterface to extend this so we can do `GI.geometry(df)` instead and have it just work, maybe we should discuss those in Jan (or earlier, if we want to move things faster?)

**Comment by alex-s-gardner** [comment]
OK, so is there a scenario where we rename GDAL "geom" columns to "geometry" for consistency, or is this bad practice?

I'm using `GI.geometrycolumns`, but it's pretty cumbersome and not easy for a new user to discover.

**Comment by rafaqz** [comment]
So there are two problems. `GI.geometrycolums` is cumbersome, and ArchGDAL is returning `:geom`.

An issue at ArchGDAL is the best place for the second problem.

The first is here, but I'm not sure how to fix that.

`GI.geomcol(t)` returning a single column is the shortest I can think of. But it really doesn't beat `:geometry`.

**Comment by evetion** [comment]
I have the same frustration, and it is an open discussion how to fix it. We went for the all should be compatible route, which now leads to some cumbersome code. I think the fixes are mostly in packages (those should use `first(GI.geometrycolumns(thing))`, not the user). And extending `GI.geometry` (now only defined for features) seems like a good plan.

**Comment by rafaqz** [comment]
`GI.geometry(sometable)` would be really nice.

But getting the column with other table syntax will also be something people try...
Is there a reason we cant also swap `:geom` to `:geometry` in ArchGDAL ?

**Comment by visr** [comment]
As I understand, `geom` doesn't come from GDAL.jl or ArchGDAL.jl, but is a default geometry column name used by some file formats. See e.g. https://gdal.org/en/latest/drivers/vector/gpkg.html

> GEOMETRY_NAME=value: Defaults to geom.

So far GDAL.jl and ArchGDAL.jl just followed GDAL's lead here. Besides `geom` vs `geometry` is also remember `` (the empty string) being one of the names that pops up. It probably makes sense for GDAL.jl to keep doing this, ArchGDAL.jl could be a bit more opinionated on this.

**Comment by rafaqz** [comment]
Yes it would make a lot of sense for ArchGDAL to change both `:geom` and empty strings to `:geometry`

**Comment by rafaqz** [comment]
Ok I made an ArchGDAL issue

**Comment by alex-s-gardner** [comment]
@evetion did we decide that this was going to be handled with in GeoDataFrames? Can this issue be closed?

**Comment by evetion** [comment]
The new GeoDataFrames will use geometry everywhere yes. Hopefully released somewhere this month.

**Comment by asinghvi17** [comment]
It looks like ArchGDAL is now one of the only packages that uses `:geom`...and we have DataAPI metadata in many places now so going from a feature collection to a DataFrame will preserve the geometry column name for `GI.geometrycolumns` to extract.
